x-pack/solutions/security/plugins/security_solution/public/data_view_manager/hooks/use_data_view.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import type { DataViewManagerScopeName } from '../constants';
+import { useFullDataView } from './use_full_data_view';
+
+/**
+ * Returns data view selection for given scopeName
+ */
+export const useDataView = (scopeName: DataViewManagerScopeName) => {
+  const { dataView, status } = useFullDataView(scopeName);
+
+  return useMemo(() => {
+    // NOTE: remove this after we are ready for undefined (lazy) data view everywhere in the app
+    // https://github.com/elastic/security-team/issues/11959
+    if (!dataView) {
+      return {
+        dataView: {
+          id: '',
+          title: '',
+        },
+        status,
+      };
+    }
+
+    return { status, dataView: dataView?.toSpec?.() };
```

Comment:
should we pass `includeFields` to the spec?

I was thinking of having `useDataView` return the full dataview, and then specify
`useDataViewSpec` or similar to specify the one with the spec

Reply:
we'll get back to this in a separate PR, skipping this would mean that the useBrowserFields should potentially be able to fetch the fields by itself when necessary (proper caching is required to do that though)